src/Rings/PBWAlgebraQuo.jl

```diff
@@ -1,5 +1,57 @@
+export PBWAlgQuo, PBWAlgQuoElem
+
+# DEVELOPER DOC  (code revised 2023-03-early)
+#  The revised code allows 2 slightly different implementations:
+#  (1) original impl where PBWAlgQuoElem is just a PBWAlgElem with a
+#      simple (almost transparent) wrapper -- all arithmetic is
+#      delegated to the PBWAlg (hence values are not automatic reduced
+#      after multiplication).
+#
+#  (2) modified impl, currently just for the case of exterior algebra.
+#      Arithmetic is delegated to the Singular impl of exterior algebra
+#      (which does automatically reduced after multiplication).  To
+#      achieve this a "new" pointer to the Singular ring is needed;
+#      this pointer is a new data field (sring) in PBQAlgQuo.
+#
+#  To preserve the original behaviour the new field "sdata" in PBWAlgQuo
+#  is set to the same value as the field "sdata" in PBWAlg (unless
+#  created by constructor for exterior_algebra.
+#
+# NOTE ABOUT PBWAlgQuoElem
+#   Values of type PBWAlgQuoElem are "wrapped" many times: the situation is
+#       an internal Singular value: a NC-polynomial & pointer to "quotient ring"
+#                                   (which contains the overlying PBWAlg in Singular
+#                                    & the reducing ideal)
+#       The NC-polynomial has a data repr which is compatible with the overlying
+#       Singular PBWAlg, so may be operated upon also by that ring (inside Singular).
+#       The Singular repr is then "wrapped" into an Oscar PBWAlgElem (which also
+#       contains a "parent" reference to the Oscar object representing the PBWAlg).
+#       This PBWAlgElem is "wrapped" once again into  PBWAlgQuoElem (which also
+#       contains a "parent" reference to the Oscar object representing the PBWAlgQuo).
+#   EXERCISE: sit down and draw a box-and-arrow diagram of values and references!!
+
+
+
 mutable struct PBWAlgQuo{T, S} <: NCRing
-  I::PBWAlgIdeal{0, T, S}
+    I::PBWAlgIdeal{0, T, S}
+    sring::Singular.PluralRing{S}  # For ExtAlg this is the Singular impl; o/w same as I.basering.sring
+
+##    # THIS DOES NOT WORK: no idea why!  [try Julia manual "parametric constructors"]
+##    function PBWAlgQuo{T, S}(I::PBWAlgIdeal{0, T, S}, sring::Singular.PluralRing{S})  where {T,S}
+##        @assert coefficient_ring(sring) == coefficient_ring(I.basering.sring) && symbols(sring) == symbols(I.basering.sring); 
+##        return new(I, sring);
+##    end
+end
+
+# For backward compatibility: ctor with 1 arg:
+#    uses "default" arith impl -- namely that from basering!
+function PBWAlgQuo(I::PBWAlgIdeal{0, T, S})  where {T, S}
+    return PBWAlgQuo{T, S}(I, I.basering.sring)
+end
+
+
+function have_special_impl(Q::PBWAlgQuo) :: Bool
```

Comment:
```suggestion
function have_special_impl(Q::PBWAlgQuo)
```
The type indicator is redundant here, Julia can infer that `!=` will produce a `Bool`